.github/workflows/reusable-test.yml

```diff
@@ -0,0 +1,157 @@
+
+      - name: Install Rosetta 2 on OS X
+        if: ${{inputs.image == 'macos-14' && inputs.architecture == 'x64'}}
+        run: /usr/sbin/softwareupdate --install-rosetta --agree-to-license
+
+      # Wine
+      - name: Setup Wine on Ubuntu
+        if: ${{runner.os == 'Linux'}}
+        uses: ./.github/actions/setup-wine-ubuntu
+
+      - name: Setup Wine on OS X
+        if: ${{runner.os == 'macOS'}}
+        uses: ./.github/actions/setup-wine-macos
+
+      - name: Get Installed Wine Information
+        if: ${{runner.os == 'Linux' || runner.os == 'macOS'}}
+        run: wine --version
+
+      - name: Get Installed Wine Programs Information
+        if: ${{runner.os == 'Linux' || runner.os == 'macOS'}}
+        run: wine uninstaller --list
+      # Wine
+
+      # .NET
+      - name: Setup .NET on Ubuntu and OS X (Intel/Arm)
+        if: ${{runner.os == 'Linux' || (inputs.image == 'macos-13' || (inputs.image == 'macos-14' && inputs.architecture == 'arm64'))}}
+        uses: ./.github/actions/setup-dotnet-unix
+        with:
+          target_framework: ''
+          target_framework_array: ${{inputs.dotnet_target_frameworks}}
+
+      - name: Setup .NET on OS X (Arm) Rosetta 2
+        if: ${{inputs.image == 'macos-14' && inputs.architecture == 'x64'}}
+        uses: ./.github/actions/setup-dotnet-macos-rosetta
+        with:
+          target_framework: ''
+          target_framework_array: ${{inputs.dotnet_target_frameworks}}
+
+      - name: Setup .NET on Windows
+        if: ${{runner.os == 'Windows'}}
+        uses: ./.github/actions/setup-dotnet-windows
+        with:
+          architecture: ${{inputs.architecture}}
+          target_framework: ''
+          target_framework_array: ${{inputs.dotnet_target_frameworks}}
+
+      - name: Get .NET path for ${{inputs.architecture}}
+        uses: ./.github/actions/get-dotnet-path
+        id: get-dotnet-path
+        with:
+          architecture: ${{inputs.architecture}}
+
+      - name: Get Installed .NET Information for ${{inputs.architecture}}
+        run: '& "${{steps.get-dotnet-path.outputs.path}}" --info'
+        shell: pwsh
+      # .NET
+
+      # Mono
+      - name: Setup Mono on Windows
+        if: ${{runner.os == 'Windows'}}
+        uses: ./.github/actions/setup-mono-windows
+        with:
+          architecture: ${{inputs.architecture}}
+
+      - name: Get Mono path for ${{inputs.architecture}}
+        uses: ./.github/actions/get-mono-path
+        id: get-mono-path
+        with:
+          architecture: ${{inputs.architecture}}
+
+      - name: Get Installed Mono Information
+        run: '& "${{steps.get-mono-path.outputs.path}}" --version'
+        shell: pwsh
+      # Mono
+
+      - name: Download Build Cache
+        uses: ./.github/actions/test-build-cache-download
+        with:
+            #image: ${{inputs.image}}
+            image: 'windows-latest'
+            architecture: ${{(inputs.architecture == 'arm64' && 'x64') || inputs.architecture}}
```

Comment:
I don't get this, what does this do? Shouldn't we be able to just pull the build cache from the same build?

Reply:
We are, but the only have Windows x86/x64 build caches. If we have OSX Arm64, it falls back to Windows x64